lib/jxl/encode.cc

```diff
@@ -1439,7 +1439,7 @@ JxlEncoderStatus JxlEncoderAddJPEGFrame(
     }
   }
   if (!io.blobs.exif.empty()) {
-    JxlOrientation orientation;
+    JxlOrientation orientation = JXL_ORIENT_IDENTITY;
```

Comment:
I don't know how relevant it is, but now what could happen is that basic info was set outside the API with some non-identity orientation, and the jpeg contains an exif, but without an orientation tag. In this case we would overwrite the user provided orientation with identity. So to be sure we could initialize this with frame_settings->enc->metadata.m.orientation.

Also I just noticed that there is already an if(!io.blobs.exif.empty()) a few lines below, maybe these should be merged.

Reply:
done